### ISSUE TITLE
Sonatype Nexus registry support

### ISSUE THREAD

**Issue by esphen** [GoogleContainerTools/jib#531]
**Description of the issue**:
This is a tracking issue/feature request for [nexus](https://www.sonatype.com/nexus-repository-sonatype) support as a registry publish target. 

**Expected behavior**:
N/A

**Steps to reproduce**: 
N/A

**Environment**: 
N/A

**Comment by coollog** [comment]
Hi @esphen , is the nexus registry currently not working for you?

**Comment by jdmarble** [comment]
It isn't working for me. I keep getting "401 Unauthorized" on `jib:build`.

I've tried using both encrypted and plaintext passwords in `~/.m2/settings.xml`. Both of those forms of credentials work with io.fabric8:docker-maven-plugin push. I've also tried removing the server element and using `docker-credential-secretservice` .

If I do `jib:dockerBuild` I can push using `docker login ...` and `docker push ...`.

**Comment by briandealwis** [comment]
Sounds like this might be #539 and so fixed in 0.9.4.

**Comment by jdmarble** [comment]
Thanks for cutting a new release so quickly! I just tried version 0.9.4 with the same result. I'm going to try some more configuration permutations to see if it is something other than nexus causing this. What I'm currently using:

```xml
<plugin>
    <groupId>com.google.cloud.tools</groupId>
    <artifactId>jib-maven-plugin</artifactId>
    <version>0.9.4</version>
    <configuration>
        <from>
            <image>${docker.private.repo}/${project.groupId}/jre:${project.version}</image>
        </from>
        <to>
            <image>${docker.private.repo}/${project.groupId}/${project.artifactId}:${containerTagSuffix}</image>
        </to>
        <allowInsecureRegistries>true</allowInsecureRegistries>
    </configuration>
</plugin>
```
The nexus registry is running insecurely over HTTP.
I really hope this isn't a PEBKAC!

**Comment by briandealwis** [comment]
I missed [this comment](https://github.com/GoogleContainerTools/jib/issues/531#issuecomment-404265193) from @jdmarble:

> The nexus registry is running insecurely over HTTP.

This is the same issue as https://github.com/GoogleContainerTools/jib/issues/545#issuecomment-404391287:
> @coollog: I realized that you were trying to use your registry with credentials over HTTP, which we don't actually support right now (you can only use credentials with registries that communicate via HTTPS). We can possibly discuss what the best way to approach this is, but for now, I would recommend setting up HTTPS on your private registry.

**Comment by esphen** [comment]
Sorry for the delay in responding to this issue, I have been swamped the last few days.

When I opened this issue, [the documentation][doc] had led me to believe that nexus was not supported. This was because I could not find anything in the documentation mentioning any repositories other than GCR, ECR and Docker Hub. Thus I opened this issue to be helpful.

I have now verified that deploying to our nexus server works as intended using the `<repository>nexus.example.com:5000/container-name</repository>` format. Note that nexus requires a port to be set.

IMHO the documentation could need a section on either specifically nexus or generally about other registries. Maybe both. Would you like to keep this issue open as a documentation issue? If not feel free to close.

Nonetheless, thanks for an excellent project, and godspeed in developing it further!

[doc]: https://github.com/GoogleContainerTools/jib/blob/v0.9.6-maven/jib-maven-plugin/README.md#configuration

**Comment by TadCordle** [comment]
We don't have anything in the issue tracker for it, but we do have a TODO in the [gradle](https://github.com/GoogleContainerTools/jib/blob/master/jib-gradle-plugin/README.md#todo-add-more-examples-for-common-registries) and [maven](https://github.com/GoogleContainerTools/jib/blob/master/jib-maven-plugin/README.md#todo-add-more-examples-for-common-registries) readmes for adding examples for other common registries. We can add something for nexus there. (Or if you'd like to contribute, you can submit a PR!)

**Comment by coollog** [comment]
Yep, we welcome contributions! For this documentation, since we are trying to migrate to a more modularized docs site, I would recommend putting the documentation under a new `.md` file in the `docs/` directory. We can then link to that where-ever applicable.

**Comment by coollog** [comment]
@danielmenezesbr @jdmarble Following up to check if you were able to get it working

**Comment by coollog** [comment]
Added to list of examples for #540. Closing as fixed unless someone is still experiencing issues.

**Comment by JaynLau** [comment]
I found that the error message is not correct. I have resoved the my problem above. 
This line message has some problem:
`com.google.cloud.tools.jib.registry.RegistryUnauthorizedException: Unauthorized for private-docker.changhongedu.cn/product-upgrade`
It points to `<to><image>private-docker.changhongedu.cn/product-upgrade</image></to>` but the authorization i configured is correct.
The error is `<from><image>hub.c.163.com/azul/zulu-openjdk:8</image></from>`, cause 'hub.c.163.com' is just a mirror for 'hub.docker.com' and the authorization of 'hub.docker.com' can not be used to 'hub.c.163.com' , so the jib:build report a "401 Unauthorized" message, but the stack trace message give me a misleading to 'private-docker.changhongedu.cn', it should be 'Unauthorized for hub.c.163.com/azul/zulu-openjdk:8', right?
